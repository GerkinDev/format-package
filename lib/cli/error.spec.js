const logErrorAndExit = require('./error');

describe('error', () => {
  const processExit = process.exit;
  let error;

  beforeEach(() => {
    jest.spyOn(console, 'error');
    console.error.mockImplementation(v => v);

    Object.defineProperty(process, 'exit', {
      value: jest.fn(),
    });

    error = new Error('Standard message');
  });

  afterEach(() => {
    Object.defineProperty(process, 'exit', { value: processExit });
    console.error.mockRestore();
  });

  it('logs generic message and exists with 1 when called without error', () => {
    logErrorAndExit();
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
    expect(process.exit).toHaveBeenCalledWith(1);
  });

  it('prints stderr', () => {
    error.stderr = 'Stderr message';
    logErrorAndExit(error);
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
  });

  it('prints stdout', () => {
    error.stdout = 'Stdout message';
    logErrorAndExit(error);
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
  });

  it('prints message', () => {
    logErrorAndExit(error);
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
  });

  it('exits with the given status or 1', () => {
    logErrorAndExit();
    logErrorAndExit({ code: 2 });

    expect(process.exit).toHaveBeenCalledTimes(2);
    expect(process.exit).toHaveBeenCalledWith(1);
    expect(process.exit).toHaveBeenCalledWith(2);
  });
});
