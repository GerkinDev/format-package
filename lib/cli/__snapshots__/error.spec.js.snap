// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error logs generic message and exists with 1 when called without error 1`] = `"[41m[37m ERROR [39m[49m Something went wrong!"`;

exports[`error prints message 1`] = `"[41m[37m ERROR [39m[49m Error: Standard message"`;

exports[`error prints stderr 1`] = `"[41m[37m ERROR [39m[49m Stderr message"`;

exports[`error prints stdout 1`] = `"[41m[37m ERROR [39m[49m Stdout message"`;
